### PR TITLE
Initial sidebar links, further setup tweaks

### PR DESCRIPTION
**Pull request by jodyheavener**
Gist: this PR adds a bunch of empty doc pages and external links for new sections in the sidebar, a basic Welcome doc, a handful of configuration changes, search capability, and some cosmetic tweaks.

Specifically:

- Adds a Welcome page detailing how to contribute from the UI
- Adds sidebar sections and external links according to the [issue](https://jira.mozilla.com/browse/EXP-1025)
  - This ticket says to create a "Mana" section, but I took a slightly different approach because _lots_ of pages link to Mana and so instead of shoehorning them into that one section I tried breaking them out into their own, as well as adding Mana to the top navbar.
- Adds some more sidebar links according to the "Existing docs" section of the [Jira Epic](https://jira.mozilla.com/browse/EXP-860)
- Removes the home page entirely, putting the app into a ["Docs-only mode"](https://v2.docusaurus.io/docs/docs-introduction#docs-only-mode), and setting the Welcome doc as the home page (now it can also be managed with Markdown!)
- Disables the dark/light mode switch, instead using the system preference
- Allow sidebar to be hide-able
- Adds Nimbus link to top navbar
- Some cosmetic changes
  - Use GitHub icon in top navbar
  - Update Favicon to be visible in light/dark browsers
  - Add dark mode logo
  - Style external sidebar links with an icon
- Adds Algolia search

### CODE REVIEW

**Comment by jodyheavener** [comment]
Before I put this up, @clouserw suggested that it might be nice to model the sidebar links using [this guidance](https://documentation.divio.com/introduction/). I like the idea of having tutorials, guides, and references, and this kind of setup could help keep things organized. Since this is a hub and there will be sections that only apply to certain groups of people (i.e. data science might not care about desktop engineer stuff), maybe these organizing categories could live underneath the links of each topic. cc @LZoog @k88hudson thoughts?

File: src/css/custom.css [hunk 1]
```diff
@@ -5,21 +5,51 @@
...
-  --ifm-color-primary-light: rgb(70, 203, 174);
-  --ifm-color-primary-lighter: rgb(102, 212, 189);
-  --ifm-color-primary-lightest: rgb(146, 224, 208);
-  --ifm-code-font-size: 95%;
 }
 
-.docusaurus-highlight-code-line {
-  background-color: rgb(72, 77, 91);
+[data-theme="dark"]:root {
+  --ifm-toc-border-color: #373737;
```
Comment (LZoog) [inline src/css/custom.css:13]: Can we lighten the link color? It appears to fail ADA color contrast in dark mode.

File: docs/welcome.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+In order to protect the main deploy branch we are requiring all content changes (editing existing pages and creating new pages) to be reviewed and merged via Pull Request. **All changes merged into the `main` branch will automatically be deployed.**
+
+### Creating new pages
+
+[Click here](https://github.com/mozilla/experimenter-docs/new/main/docs) to start creating a new Markdown file in the repository. Once you've written your content you can can click "Commit Changes" to create a new branch and be presented with the option to open a Pull Request.
+
+:::important
+
+One caveat to adding new pages is that if you want them to show up in the sidebar they must be added to the [sidebars.js](https://github.com/mozilla/experimenter-docs/blob/main/sidebars.js) file. If you are adding a new file using the GitHub UI you must first make sure you have changed to the branch you just created with your new page commit. Then you can use the UI to add your new page to the sidebar and add that commit to the Pull Request.
```
Comment (LZoog) [inline docs/welcome.md:44]: Wellllll... I could have sworn Docusaurus did this automagically. ☹️ I'll figure this out when working on the maintenance workflow docs (you've already written like more than half of it so thanks, ha).

File: sidebars.js [hunk 3]
```diff
@@ -1,6 +1,91 @@
...
+      items: [
+        "desktop-engineers-root",
+        {
+          type: "link",
+          label: "Front-End Migration Guide",
+          href: "https://www.notion.so/Nimbus-Migration-Guide-for-Desktop-Front-End-Experiments-d36c21e505f84550aad1202897fc4ba3",
+        },
+        {
+          type: "link",
+          label: "SDK Docs (draft)",
```
Comment (LZoog) [inline sidebars.js:80]: Will you put `(X-Man)` in parenthesis here too?

**Review by LZoog — APPROVED** [review]
This LGTM! I double checked all the links were there, and I think this setup/organization is great👌

**Comment by jodyheavener** [comment]
> I like the way this looks as well, though is it a little premature to make it this layered with the amount of docs that we have planned right now? Maybe we could create a follow up ticket for this? It looks great how you have it in this PR IMO with the collapsible sections. :)

Thanks for your input here! I do like the suggested way of organizing into references, guides, tutorials, etc. but you might be right that it's early for this. I've created https://jira.mozilla.com/browse/EXP-1048 so we can follow up and maybe try a re-organization once we have more content. cc @clouserw

**Pull request merged**